acm-uic/IaC
# 😎 Feature/Argocd setup

This PR installs ArgoCD onto the on-prem Kubernetes cluster. No endpoint has been exposed for this application yet nor has any connection with Github been established yet.

If you need to access the webUI, that can be done via a kubectl port-forward command: `kubectl --kubeconfig k8s-kubeconfig -n argocd port-forward service/argocd-release-server 8080:80`

**DO NOT** manually deploy applications via the webUI or risk it being destroyed. All deployments should be done via IaC.


No automation is running this Terraform currently, so it is currently being manually run on an adhoc basis. This will change later once we install Github runners onto the cluster.

### Deployment output
```bash
[1_k8s_cluster_setup]$ terraform apply -var-file kubernetes/terraform/stacks/1_k8s_cluster_setup/configurations/acmapp_prod.tfvars

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # helm_release.argocd will be created
  + resource "helm_release" "argocd" {
      + atomic                     = false
      + chart                      = "argo-cd"
      + cleanup_on_fail            = false
      + create_namespace           = false
      + dependency_update          = false
      + disable_crd_hooks          = false
      + disable_openapi_validation = false
      + disable_webhooks           = false
      + force_update               = false
      + id                         = (known after apply)
      + lint                       = false
      + manifest                   = (known after apply)
      + max_history                = 0
      + metadata                   = (known after apply)
      + name                       = "argocd-release"
      + namespace                  = "argocd"
      + pass_credentials           = false
      + recreate_pods              = false
      + render_subchart_notes      = true
      + replace                    = false
      + repository                 = "https://argoproj.github.io/argo-helm"
      + reset_values               = false
      + reuse_values               = false
      + skip_crds                  = false
      + status                     = "deployed"
      + timeout                    = 300
      + verify                     = false
      + version                    = "5.37.1"
      + wait                       = true
      + wait_for_jobs              = false

      + set {
          + name  = "configs.params.server\\.insecure"
          + value = "true"
        }
      + set {
          + name  = "redis-ha.enabled"
          + value = "true"
        }
      + set {
          + name  = "repoServer.autoscaling.enabled"
          + value = "true"
        }
      + set {
          + name  = "server.autoscaling.enabled"
          + value = "true"
        }
      + set {
          + name  = "server.service.annotations.external-dns\\.alpha\\.kubernetes\\.io/hostname"
          + value = "cd.acmuic.org"
        }
    }

  # kubernetes_namespace.argocd_namespace will be created
  + resource "kubernetes_namespace" "argocd_namespace" {
      + id = (known after apply)

      + metadata {
          + generation       = (known after apply)
          + name             = "argocd"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

kubernetes_namespace.argocd_namespace: Creating...
kubernetes_namespace.argocd_namespace: Creation complete after 0s [id=argocd]
helm_release.argocd: Creating...
helm_release.argocd: Still creating... [10s elapsed]
helm_release.argocd: Still creating... [20s elapsed]
helm_release.argocd: Still creating... [30s elapsed]
helm_release.argocd: Still creating... [40s elapsed]
helm_release.argocd: Still creating... [50s elapsed]
helm_release.argocd: Still creating... [1m0s elapsed]
helm_release.argocd: Still creating... [1m10s elapsed]
helm_release.argocd: Still creating... [1m20s elapsed]
helm_release.argocd: Still creating... [1m30s elapsed]
helm_release.argocd: Still creating... [1m40s elapsed]
helm_release.argocd: Still creating... [1m50s elapsed]
helm_release.argocd: Still creating... [2m0s elapsed]
helm_release.argocd: Still creating... [2m10s elapsed]
helm_release.argocd: Still creating... [2m20s elapsed]
helm_release.argocd: Still creating... [2m30s elapsed]
helm_release.argocd: Still creating... [2m40s elapsed]
helm_release.argocd: Still creating... [2m50s elapsed]
helm_release.argocd: Still creating... [3m0s elapsed]
helm_release.argocd: Still creating... [3m10s elapsed]
helm_release.argocd: Still creating... [3m20s elapsed]
helm_release.argocd: Creation complete after 3m27s [id=argocd-release]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```
